saleor/product/models/base.py

```diff
@@ -156,54 +206,79 @@ class Meta:
         verbose_name_plural = pgettext_lazy('Product model', 'products')
 
     def __iter__(self):
+        # type: () -> Iterable[django.db.models.QuerySet]
         if not hasattr(self, '__variants'):
             setattr(self, '__variants', self.variants.all())
         return iter(getattr(self, '__variants'))
 
     def __repr__(self):
+        # type: () -> str
```

Comment:
Is it really useful to add types to magic methods with a predefined interface? You never call `__repr__` directly and `repr` is a builtin typed to return a string.